l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -228,16 +233,21 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
     /// @dev for backwards compatibility and to automatically register l1 assets, and we return the correct info.
     function _getAssetProperties(bytes32 _assetInfo) internal returns (address l1Asset, bytes32 assetInfo) {
         l1Asset = assetAddress[_assetInfo];
-        assetInfo = _assetInfo;
         if (l1Asset == address(0) && (uint256(_assetInfo) <= type(uint160).max)) {
             l1Asset = address(nativeTokenVault);
             assetInfo = keccak256(abi.encode(block.chainid, NATIVE_TOKEN_VAULT_VIRTUAL_ADDRESS, _assetInfo));
+            assetAddress[_assetInfo] = l1Asset;
```

Comment:
This is a getter function, it should now set anything, no?

Reply:
you're right, I just thought it's better to update mapping, so on 2nd read it's more efficient

Reply:
It happens via the nativeTokenVault.registerToken, so  there is no need for it

Reply:
updated